src/modules/billing/services/supplementary-billing-service/supplementary-processor.js

```diff
@@ -47,6 +47,14 @@ const commonTransactionKeys = [
   'financialYearEnding'
 ]
 
+const pick = (obj, props) => {
```

Comment:
I chose to add a function here for pick otherwise I would have had to duplicate the commonTransactionKeys and abstractionPeriodKeys which would have been an additional 24 lines of code (since pick is used three times in this file)

Reply:
Again, I think the way those key arrays are being used means we can use them without needing to repeat them and avoid having to implement a `pick()` method within the module.